org.omg.sysml/src/org/omg/sysml/util/repository/APIModel.java

```diff
@@ -0,0 +1,253 @@
+	}
+	
+	/**
+	 * Adds a proxy object for every reference that cannot be resolved in the scope
+	 * of this model
+	 * 
+	 * The API server doesn't allow unresolvable references but the current
+	 * implementation doesn't support used projects yet. Since every model
+	 * references at least the standard library it is unavoidable to end up with
+	 * references that point outside the API model. This method serves as a
+	 * workaround for this issue by inserting proxy objects for out of scope references.
+	 */
+	public void addOutOfScopeReferencesAsProxies() {
+		Set<LocalReference> localReferences = modelElements.values().stream()
+			.flatMap(el -> el.values().stream())
+			.flatMap(this::flatten)
+			.collect(Collectors.toSet());
+		
+		Set<UUID> added = new HashSet<>();
+		for (LocalReference ref: localReferences) {
+			UUID uuid = ref.getUUID();
+			if (getElement(uuid) == null && !added.contains(uuid)) {
+				Element proxy = new Element();
+				proxy.put(ProjectRepository.ID_FIELD, uuid.toString());
+				proxy.put("@type", ref.getType());
+				proxy.put("isLibraryElement", ref.isLibraryElement());
+				addModelRoot(uuid, proxy);
+			}
+		}
+	}
+	
+	private Stream<LocalReference> flatten(Object object) {
+		if (object instanceof LocalReference ref) {
+			return Stream.of(ref);
+		} else if (object instanceof Collection<?> col) {
+			return col.stream().filter(LocalReference.class::isInstance)
+					.map(LocalReference.class::cast);
+		} else {
+			return Stream.empty();
+		}
+	}
+	
+	/**
+	 * Creates a JSON representation of the model.
+	 * 
+	 * @return JSON representation
+	 */
+	public String toJson(Gson gson) {
+		return gson.toJson(getModelElements().values());
+	}
+	
+	/**
+	 * Creates a JSON tree representation of the model.
+	 * 
+	 * @return JSON tree representation
+	 */
+	public JsonElement toJsonTree(Gson gson) {
+		return gson.toJsonTree(getModelElements().values());
+	}
+	
+	/**
+	 * API model compatible representation of model references
+	 * This class is part of a workaround which requires
+	 * proxy creation for unresolved references. Objects of this class also store the type of the referenced
+	 * element (transient only). This is then used by {@link APIModel#addOutOfScopeReferencesAsProxies()}
+	 * to insert proxies into the API model.
+	 */
+	@SuppressWarnings("serial")
+	public static class LocalReference extends HashMap<String, UUID> {
+		
+		private transient String type;
+		private boolean isLibraryElement;
+
+		public LocalReference(UUID identifier, String type, boolean isLibraryElement) {
+			put(ProjectRepository.ID_FIELD, identifier);
+			this.type = type;
+			this.isLibraryElement = isLibraryElement;
+		}
+		
+		public String getType() {
+			return type;
+		}
+		
+		public boolean isLibraryElement() {
+			return isLibraryElement;
+		}
+
+		public UUID getUUID() {
+			return get(ProjectRepository.ID_FIELD);
+		}
+		
+		@Override
+		public boolean equals(Object o) {

```

Comment:
When overriding Object#equals, also update hashcode as well to make sure that the generic contract of that method. Failing to do so could cause issues e.g. when using the objects in HashSets and HashMaps